### ISSUE TITLE
Questions on Bounding Box Coordinate Format and Image Resizing for Qwen3-VL Fine-tuning

### ISSUE THREAD

**Issue by Asunatan** [QwenLM/Qwen3-VL#1486]
Hi there,

I’m a beginner working with Qwen-VL models, and after reading the **Qwen3-VL** blog post, I have two questions regarding how to prepare bounding box training data — both centered around coordinate formatting and image preprocessing.
1. I noticed that **Qwen3-VL** appears to use bounding box coordinates in a relative format with range [0, 1000], similar to Qwen2-VL. If I want to prepare my own custom training data, should I follow the same conversion logic as implemented in [qwen-vl-finetune/tools/process_bbox.ipynb](https://github.com/QwenLM/Qwen3-VL/blob/main/qwen-vl-finetune/tools/process_bbox.ipynb), specifically the function `convert_to_qwen2vl_format(bbox, h, w)`? Is this conversion code still applicable and recommended for Qwen3-VL?
2. Assuming we convert coordinates to the [0, 1000] relative scale — if the model internally applies `smart resize` (or any dynamic resizing) to input images during training, do we need to adjust the coordinates a **second time** after resizing (i.e., perform a **double conversion** to maintain the [0, 1000] scale), or does the model handle this automatically?

I’d really appreciate any guidance or clarification — thank you in advance!

**Comment by tmliang** [comment]
+1

**Comment by HumanZhong** [comment]
Hi, 
1. Yes, you can refer to `convert_to_qwen2vl_format` to make relative coordinates ranging from 0 to 1000.
2. Since we use relative coordinates, resizing image will not change the target coordinates. For example, a point (25, 25) on a 100x100 image, no matter how you resize this image, the target coordinates will always be (250, 250). After you get the model response, you just need to do such conversion: `img_x = output_x / 1000 * img_w` for further visualization.
Hope this message helps you. If you still have any questions, please feel free to ask!

**Comment by Asunatan** [comment]
Thanks a lot for the clear explanation! Your example made the coordinate system totally intuitive, and the conversion formula will save me a ton of time. Really appreciate your quick help.

**Comment by zhouxinwen-cpu** [comment]
For qwen 3 vl， is the height and width of input images  multiples  of 32?  If it is, How can the height of input image  be  multiples  of 32 and at the same time bbox coordinate range from 0 to 1000?

**Comment by HumanZhong** [comment]
> For qwen 3 vl， is the height and width of input images multiples of 32? If it is, How can the height of input image be multiples of 32 and at the same time bbox coordinate range from 0 to 1000?

@zhouxinwen-cpu 
Hi, here is an example: 
1. a 100x100 image, and the original bbox coordinates is (25, 25, 50, 50);
2. when you feed this image into the model, the model will first resize it into 96x96 because 96 is the multiple of 32 closest to 100.
3. The gt bbox is always (250, 250, 500 500) [because it is rescaled into [0, 1000]], EVEN IF THE IMAGE IS RESIZED.
4. Thus, the gt bbox is always fixed. You can even resize it from 100x100 to 3200x3200, the gt bbox should always be [250, 250, 500, 500]

Hope this message helps you. If you still have any questions, please feel free to ask!

**Comment by Rakshith12-pixel** [comment]
@Asunatan Just curious, do you have a fine-tuning script for Qwen3 VL??

**Comment by Asunatan** [comment]
@Rakshith12-pixel 
Hi! You can directly use the official Qwen3-VL source—it's extremely helpful.
For a quick start, just call `apply_chat_template `on your conversations, then grab everything between `<|im_start|>assistant\n` and `<|im_end|>\n` to build your labels (keep the `<|im_end|>\n` so the model learns to auto-stop).
After that, plug everything into the vanilla 🤗 Trainer:
```python
trainer = Trainer(
    model=model,
    args=training_args,
    data_collator=data_collator,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
)
```
For the recommended data format, please refer to:[https://github.com/QwenLM/Qwen3-VL/issues/1513](https://github.com/QwenLM/Qwen3-VL/issues/1513)

**Comment by ireneYang** [comment]
> Hi, here is an example:
> 
> 1. a 100x100 image, and the original bbox coordinates is (25, 25, 50, 50);
> 2. when you feed this image into the model, the model will first resize it into 96x96 because 96 is the multiple of 32 closest to 100.
> 3. The gt bbox is always (250, 250, 500 500) [because it is rescaled into [0, 1000]], EVEN IF THE IMAGE IS RESIZED.
> 4. Thus, the gt bbox is always fixed. You can even resize it from 100x100 to 3200x3200, the gt bbox should always be [250, 250, 500, 500]
> 
> Hope this message helps you. If you still have any questions, please feel free to ask!
When I use 2d_grounding.ipynb in codebook to dectect and get bbox of  CAN and paper cup and fruit from image spatio_case2_plan.png (assets/spation_understanding), then I  anti-transform (0~1000) the bbox  to the original image (400*229) space , and draw the rectangle on the original image, the rectangle could not fit the objects ( can, paper cup ,and fruit 

<img width="400" height="225" alt="Image" src="https://github.com/user-attachments/assets/7647f98f-9d81-416f-a3e5-f3da9d36cfda" />

how can I fix this issue?  I used to use upscale and float presize method to transform the bbox cordination,  it couldnot work
